MSAL/src/MSALErrorConverter.m

```diff
@@ -174,7 +174,13 @@ + (NSError *)errorWithDomain:(NSString *)domain
         
         if (![s_recoverableErrorCode containsObject:mappedCode])
         {
-            internalCode = mappedCode;
+            // If mapped code is MSALErrorInternal, set internalCode to nil
+            // to avoid the case when both mapped and internal code are MSALErrorInternal.
```

Comment:
not sure about this one. This means developer will encounter nil (or 0 since they translate it to integer), and will be an unexpected error for them. Should we instead introduce a new internal error (e.g. unexpected error)?